java/org/apache/juli/FileHandler.java

```diff
@@ -585,4 +517,34 @@ private String obtainDateFromPath(Path path) {
             return null;
         }
     }
+
+
+    static class AsyncTaskExecutor {
+
+        private Thread thread;
+        private BlockingQueue<Runnable> tasks;
+
+        AsyncTaskExecutor() {
+            this.tasks = new LinkedBlockingQueue<>();
+            this.thread = new Thread(() -> {
+                try {
```

Comment:
This should probably be wrapped in a `while (true)` loop. At the moment it will execute just the first task and finish.